src/pages/write-smart-contracts/billboard-tutorial.md

```diff
@@ -0,0 +1,281 @@
+
+- Set up a development environment with Clarinet
+- Define codes for error handling
+- Add a data storage variable with functions to get and set the variable
+- Add a STX transfer function within the variable setter
+- Develop a unit test to verify the contract works as expected
+
+The [final code for this tutorial][] is available in the Clarinet repository.
+
+## Prerequisites
+
+For this tutorial, you should have a local installation of [Clarinet][]. Refer to [Installing Clarinet][] for
+instructions on how to set up your local environment. You should also have a text editor or IDE to edit the Clarity
+smart contract.
+
+For developing the unit test, it's recommended that you have an IDE with Typescript support, such as
+[Visual Studio Code][].
+
+## Step 1: set up the project
+
+With Clarinet installed locally, open a new terminal window and create a new Clarinet project. Add a smart contract and
+an empty test file to the project:
+
+```sh
+clarinet new billboard-clarity && cd billboard-clarity
+clarinet contract new billboard
+```
+
+These commands create the necessary project structure and contracts for completing this tutorial. Remember that at
+any point during this tutorial you can use `clarinet check` to check the validity of your Clarity syntax.
+
+## Step 2: create message storage
+
+Open the `contracts/billboard.clar` file in a text editor or IDE. For this tutorial, you'll use the boilerplate comments
+to structure your contract for easy readability.
+
+In this step, you'll add a variable to the contract that stores the billboard message, and define a getter function to
+read the value of the variable.
+
+Under the `data maps and vars` comment, define the `billboard-message` variable. Remember that you must define the type of
+the variable, in this case `string-utf8` to support emojis and extended characters. You must also define the
+maximum length of the variable, for this tutorial use the value `500` to allow for a longer message. You must also
+define the initial value for the variable.
+
+```clarity
+;; data maps and vars
+(define-data-var billboard-message (string-utf8 500) u"Hello world!")
+```
+
+You also should define a read-only getter function returns the value of the `billboard-message` variable.
+
+```clarity
+;; public functions
+(define-read-only (get-message)
+    (var-get billboard-message))
+```
+
+These are the required methods for storing and accessing the message on the billboard.
+
+## Step 3: define set message function
+
+Define a method to set the billboard message. Under the public functions, define a `set-message` function. This public
+function takes a `string-utf8` with a max length of `500` as the only argument. Note that the type of the argument
+matches the type of the `billboard-message` variable. Clarity's type checking ensures that an invalid input to the
+function doesn't execute.
+
+```clarity
+(define-public (set-message (message (string-utf8 500)))
+    (var-set billboard-message message)
+)
```

Comment:
I think the closing bracket should be on the line before. I also get an error:

```
public functions must return an expression of type 'response', found 'bool'clarity
```

We should probably wrap it into `ok`?

Reply:
This code works as written in my Clarinet console. Also the `var-set` functions handles the return of `ok`.

Reply:
it returns a bool type: https://blockstack-docs-git-docs-billboard-tutorial-blockstack.vercel.app/references/language-functions#var-set

Reply:
This is totally my bad - I thought we were talking about Reed's function, but this is my simplified one. It should be fixed to work now.